src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DiagnosticReportStatus.java

```diff
@@ -0,0 +1,73 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * Example STS++UN'
+ */
+@EqualsAndHashCode(callSuper = false)
+@Builder
+@Getter
+@AllArgsConstructor
+public class DiagnosticReportStatus extends Segment {
+
+    private static final String KEY = "STS";
+
+    private final String detail;
+    private final @NonNull ReportStatusCode event;
+
+    public static DiagnosticReportStatus fromString(final String edifactString) {
+        if (!edifactString.startsWith(KEY)) {
+            throw new IllegalArgumentException("Can't create " + DiagnosticReportStatus.class.getSimpleName() + " from " + edifactString);
+        }
+        String detail = Split.byPlus(edifactString)[1];
+        String event = Split.byPlus(edifactString)[2];
+        return new DiagnosticReportStatus(detail, ReportStatusCode.fromCode(event));
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        if (detailsEmpty()) {
+            return event.getCode();
+        } else {
+            return detail + "+" + event.getCode();
+        }
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (event.getCode().isBlank()) {
+            throw new EdifactValidationException(getKey() + ": Status is required");
+        }
+    }
+
+    @Override
+    public String toEdifact() throws EdifactValidationException {
+        super.validate();
+        if (detailsEmpty()) {
```

Comment:
Apache's `StringUtils.isEmpty(cs)` covers both null and empty strings - no need to reinvent the wheel here
```suggestion
        if (StringUtils.isEmpty(detail)) {
```

Reply:
Also applies to the `getValue()` method here.